### PR TITLE
Fatotesa - custom asymmetric split keyboard 

### PR DESCRIPTION
**Pull request by stefangluszek**
## Description
My custom split keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/fatotesa/config.h
Comment (waffle87) [inline keyboards/fatotesa/config.h:1]: The contents of this file should be migrated to `info.json`, and this file removed.

File: keyboards/fatotesa/info.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "split": {
+        "enabled": true,
+        "transport": {
+            "protocol": "serial"
+        },
```
Comment (waffle87) [inline keyboards/fatotesa/info.json:19]:
```suggestion
```

File: keyboards/fatotesa/info.json [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                    }
+                ]
+            }
+        }
+    },
+    "matrix_pins": {
+        "cols": ["F5", "F6", "F7", "B1", "B3", "B2", "B6", null],
+        "rows": ["D4", "C6", "D7", "E6", "B4", "B5"]
+    },
+    "processor": "atmega32u4",
```
Comment (waffle87) [inline keyboards/fatotesa/info.json:42]:
```suggestion
```

File: keyboards/fatotesa/readme.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# fatotesa
+
+![fatotesa](https://fatotesa.se/assets/images/Fatotesa.png)
```
Comment (waffle87) [inline keyboards/fatotesa/readme.md:3]: This image should be rehosted on Imgur.

File: keyboards/fatotesa/rules.mk [hunk 4]
```diff
@@ -0,0 +1,3 @@
+ENCODER_ENABLE = yes
+CAPS_WORD_ENABLE = yes
+ENABLE_BOOTMAGIC = yes
```
Comment (waffle87) [inline keyboards/fatotesa/rules.mk:3]:
```suggestion
# This file intentionally left blank
```

File: keyboards/fatotesa/info.json [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    "keyboard_name": "fatotesa",
+    "maintainer": "stefangluszek",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (waffle87) [inline keyboards/fatotesa/info.json:13]:
```suggestion
        "nkro": true,
        "encoder": true,
        "caps_word": true
```

File: keyboards/fatotesa/info.json [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                {"matrix": [0, 0], "label":"esc", "x":0, "y":0},
+                {"matrix": [0, 2], "label":"f1", "x":2, "y":0},
+                {"matrix": [0, 3], "label":"f2", "x":3, "y":0},
+                {"matrix": [0, 4], "label":"f3", "x":4, "y":0},
+                {"matrix": [0, 5], "label":"f4", "x":5, "y":0},
+
+                {"matrix": [6, 4], "label":"end", "x":13.25, "y":0},
+                {"matrix": [6, 5], "label":"ins", "x":14.25, "y":0},
+                {"matrix": [6, 6], "label":"del", "x":15.25, "y":0},
+                {"matrix": [6, 7], "label":"<i class='fa fa-repeat'></i>", "x":16.5, "y":0, "encoder": 0},
```
Comment (waffle87) [inline keyboards/fatotesa/info.json:61]:
```suggestion
                {"matrix": [6, 7], "label":"enc", "x":16.5, "y":0, "encoder": 0},
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/fatotesa/config.h
Comment (stefangluszek) [reply keyboards/fatotesa/config.h:1]: How do I configure
```c
#define ENCODERS_PAD_A { }
#define ENCODERS_PAD_B { }
```
Is this the correct way:
```json
"encoder": {
    "rotary": []
},
```

**Comment by stefangluszek** [comment]
@waffle87 Thank you for the review, I've pushed fixes. I wasn't sure about the encoder config in info.json if I only have encoder on the right side.

File: keyboards/fatotesa/config.h
Comment (waffle87) [inline keyboards/fatotesa/config.h:1]: This file can be removed.

File: keyboards/fatotesa/info.json [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    },
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0000",
+        "vid": "0xFEED"
+    },
+    "caps_word": {
+        "enabled": true
+    },
```
Comment (waffle87) [inline keyboards/fatotesa/info.json:60]:
```suggestion
```

File: keyboards/fatotesa/readme.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# fatotesa
+
+![fatotesa](https://imgur.com/LbxQcU2)
```
Comment (waffle87) [inline keyboards/fatotesa/readme.md:3]:
```suggestion
![fatotesa](https://i.imgur.com/LbxQcU2.png)
```

File: keyboards/fatotesa/info.json [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+        "console": false,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "encoder": true,
+        "caps_word": true
+    },
+    "encoder": {
+        "rotary": []
```
Comment (waffle87) [inline keyboards/fatotesa/info.json:19]: This'll be fine knowing you only have an encoder on the right half. The right half specific settings are more so for if the right half encoder is using different pins.
```suggestion
        "rotary": [
            { "pin_a": "D1", "pin_b": "D0" }
        ]
```

File: keyboards/fatotesa/info.json [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "encoder": true,
```
Comment (waffle87) [inline keyboards/fatotesa/info.json:15]: Duplicate from above.
```suggestion
```

File: keyboards/fatotesa/info.json [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+        "encoder": {
+            "right": {
+                "rotary": [
+                    {
+                        "pin_a": "D1",
+                        "pin_b": "D0"
+                    }
+                ]
+            }
+        },
```
Comment (waffle87) [inline keyboards/fatotesa/info.json:43]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**